common/app/pagepresser/HtmlCleaner.scala

```diff
@@ -197,7 +197,7 @@ abstract class HtmlCleaner extends GuLogging {
   }
 
   private def secureSource(src: String): String = {
-    src.replaceAllLiterally("http://", "//")
+    src.replace("http://", "//")
```

Comment:
should this be replaceAll like in `StringEncodings.scala` - only asking because the original method is a type of `replaceAll` - unlikely that there will be more than one "http://" but just in case?

Reply:
Thanks for your suggestion! 

### Short version
We're using the recommendation of the error here so `replace` is what we need:

```
[error] /Users/Ioanna_Kokkini/Projects/frontend/common/app/pagepresser/HtmlCleaner.scala:200:9: method replaceAllLiterally in class StringOps is deprecated (since 2.13.2): Use `s.replace` as an exact replacement
[error]     src.replaceAllLiterally("http://", "//")
[error]         ^
```
See also: https://www.scala-lang.org/api/current/scala/collection/StringOps.html in section `Deprecated Value Members` def replaceAllLiterally`.

### Long version (feel free to skip! 🙂)
I took your suggestion as an opportunity to understand this a bit better and here is what I found:
* `src` is of type [`java.lang.String`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html) so it has access to all its methods.
* But it is _also_ a [`scala.collection.immutable.StringOps`](https://www.scala-lang.org/api/current/scala/collection/StringOps.html) and has access to some extra methods. `replaceAllLiterally` is one of them.
* How is it possible to be both a Java String and a Scala StringOps? Scala offers the magic of _implicit conversions_, which "give the Scala compiler the ability to convert one type into another" (I'm like 🤯 but that's the way it is.)
* So what happened here is `replaceAllLiterally` of `scala.collection.immutable.StringOps` got deprecated but we can use `replace` (of `java.lang.String`) "as an exact replacement".
* Why `replace` and not `replaceAll`?   
From the docs [`replace`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html) is actually doing the same thing as [`replaceAllLiterally`](https://www.scala-lang.org/api/current/scala/collection/StringOps.html), they replace all the occurrences of a substring, whereas [`replaceAll`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/String.html) replaces all the occurrences of a given regex.

Here is a [great explanation of how implicit conversion works on a String](https://www.oreilly.com/library/view/scala-cookbook/9781449340292/ch01s01.html) and a [more in depth tutorial of how implicit conversions work](https://www.baeldung.com/scala/implicit-conversions).